rpcs3/Emu/RSX/D3D12/D3D12GSRender.cpp

```diff
@@ -243,7 +244,7 @@ D3D12GSRender::D3D12GSRender()
 
 	m_rtts.init(m_device.Get());
 	m_readback_resources.init(m_device.Get(), 1024 * 1024 * 128, D3D12_HEAP_TYPE_READBACK, D3D12_RESOURCE_STATE_COPY_DEST);
-	m_buffer_data.init(m_device.Get(), 1024 * 1024 * 896, D3D12_HEAP_TYPE_UPLOAD, D3D12_RESOURCE_STATE_GENERIC_READ);
+	m_buffer_data.init(m_device.Get(), vram_size - 1024 * 1024 * 128, D3D12_HEAP_TYPE_UPLOAD, D3D12_RESOURCE_STATE_GENERIC_READ);
```

Comment:
It's in bytes: https://msdn.microsoft.com/ru-ru/library/windows/desktop/bb173058(v=vs.85).aspx
@raven02 probably you'd want to check with SharedMemory too for iGPUs. But the issue stands - I tried `Sonic CD` and `fw_dice_test` and the both won't run regarding of the adapter I select.
Btw, how does your code handle WARP?